nearpy/tests/engine_tests.py

```diff
@@ -73,5 +74,53 @@ def test_retrieval_sparse(self):
             self.assertEqual(y_data, x_data)
             self.assertAlmostEqual(y_distance, 0.0, delta=delta)
 
+    def get_keys(self, engine):
+        for lshash in engine.lshashes:
+            bucket = engine.storage.buckets[lshash.hash_name][lshash.hash_name]
+            return {i[1] for i in bucket}
```

Comment:
You return keys from the first bucket of the first hash and it makes your second test really weird. If you want to get all data stored in the engine you should union content of all buckets for all hashes.